sonic_package_manager/service_creator/creator.py

```diff
@@ -476,28 +495,143 @@ def set_initial_config(self, package):
             cfg = conn.get_config()
             new_cfg = init_cfg.copy()
             utils.deep_update(new_cfg, cfg)
+            self.validate_config(new_cfg)
             conn.mod_config(new_cfg)
 
     def remove_config(self, package):
-        """ Remove configuration based on init-cfg tables, so having
-        init-cfg even with tables without keys might be a good idea.
-        TODO: init-cfg should be validated with yang model
-        TODO: remove config from tables known to yang model
+        """ Remove configuration based on package YANG module.
 
         Args:
             package: Package object remove initial configuration for.
         Returns:
             None
         """
 
-        init_cfg = package.manifest['package']['init-cfg']
-        if not init_cfg:
+        if not package.metadata.yang_module_str:
             return
 
-        for conn in self.sonic_db.get_connectors():
-            for table in init_cfg:
-                for key in init_cfg[table]:
-                    conn.set_entry(table, key, None)
+        module_name = self.cfg_mgmt.get_module_name(package.metadata.yang_module_str)
+        for tablename, module in self.cfg_mgmt.sy.confDbYangMap.items():
+            if module.get('module') != module_name:
+                continue
+
+            for conn in self.sonic_db.get_connectors():
```

Comment:
Curious! We deal with CONFIG-DB only. If so, why this for loop?

Reply:
We deal with CONFIG-DB, /etc/sonic/config_db.json and /etc/sonic/init_cfg.json.